diff --git a/greetd/src/session/worker.rs b/greetd/src/session/worker.rs
index 5c145bd..8929da3 100644
--- a/greetd/src/session/worker.rs
+++ b/greetd/src/session/worker.rs
@@ -225,6 +225,7 @@ fn worker(sock: &UnixDatagram) -> Result<(), Error> {
     _ = pam.putenv(&"XDG_SESSION_CLASS");
 
     // Prepare some strings in C format that we'll need.
+    let user_name = (&user.name).clone();
     let cusername = CString::new(user.name)?;
     let command = if source_profile {
         format!(
@@ -259,8 +260,9 @@ fn worker(sock: &UnixDatagram) -> Result<(), Error> {
             prctl(PrctlOption::SET_PDEATHSIG(libc::SIGTERM)).expect("unable to set death signal");
 
             // Change working directory
+            let user_dir = user.dir.clone();
             if let Err(e) = env::set_current_dir(user.dir) {
-                eprintln!("unable to set working directory: {e}");
+                eprintln!("unable to set working directory '{}' for user {}: {e}", user_dir.to_string_lossy(), user_name);
             }
 
             // Run